### ISSUE TITLE
How to overwrite head module

### ISSUE THREAD

**Issue by Yoline777** [facebookresearch/detectron2#1144]
I try to add a simple head module, but get error:
Loss became infinite or NaN

```
@META_ARCH_REGISTRY.register()
class TestArch(nn.Module):
    def __init__(self, cfg):
        super().__init__()

        self.device = torch.device(cfg.MODEL.DEVICE)
        self.backbone = build_backbone(cfg) #build_resnet_fpn_backbone
        self.roi_heads = TestHeads(cfg, self.backbone.output_shape())

        assert len(cfg.MODEL.PIXEL_MEAN) == len(cfg.MODEL.PIXEL_STD)
        num_channels = len(cfg.MODEL.PIXEL_MEAN)
        pixel_mean = torch.Tensor(cfg.MODEL.PIXEL_MEAN).to(self.device).view(num_channels, 1, 1)
        pixel_std = torch.Tensor(cfg.MODEL.PIXEL_STD).to(self.device).view(num_channels, 1, 1)
        self.normalizer = lambda x: (x - pixel_mean) / pixel_std
        self.to(self.device)

    def forward(self, batched_inputs):
        images = self.preprocess_image(batched_inputs)
        features = self.backbone(images.tensor)
        _, detector_losses = self.roi_heads(images, features, None, gt_instances)

        losses = {}
        losses.update(detector_losses)
        return losses

    def preprocess_image(self, batched_inputs):
        """
        Normalize, pad and batch the input images.
        """
        images = [x["image"].to(self.device) for x in batched_inputs]
        images = [self.normalizer(x) for x in images]
        images = ImageList.from_tensors(images, self.backbone.size_divisibility)
        return images

class TestHeads(nn.Module):
    def __init__(self, cfg, input_shape):
        super().__init__()
        self.in_features = cfg.MODEL.KEYPOINT_MASK.IN_FEATURES

    def forward(self, images, features, proposals, targets=None):
        features = [features[f] for f in self.in_features]
        loss = 0
        for feature in features:
            loss += torch.sum(feature)**2
        return [], {'loss': loss}
```

**Comment by ppwwyyxx** [comment]
As the [issue template](https://github.com/facebookresearch/detectron2/blob/master/.github/ISSUE_TEMPLATE/unexpected-problems-bugs.md) mentions:

> If you expect the model to converge / work better, note that we do not give suggestions on how to train a new model. Only in one of the two conditions we will help with it: (1) You're unable to reproduce the results in detectron2 model zoo. (2) It indicates a detectron2 bug.

We provide configs & models with standard academic settings and expect users to have the knowledge to choose or design appropriate models & parameters for their own tasks.
